File: JavaDemoGithubPractice/app/src/main/java/com/example/javademogithubpractice/room/dao/AuthUserDao.java
package com.example.javademogithubpractice.room.dao;



import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.example.javademogithubpractice.room.model.AuthUser;
import java.util.List;
import io.reactivex.Completable;
import io.reactivex.Observable;

@Dao
public interface AuthUserDao {

    @Insert
    Completable insertAuthUser(AuthUser authUser);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    Completable updateAuthUser(AuthUser authUser);

    @Query("SELECT * FROM authUser_database")
    Observable<List<AuthUser>> loadAll();
//
//    @Query("DELETE FROM authUser_database WHERE accessToken = :accessAuthUser")
//    Completable deleteAuthUser(String accessAuthUser);

    @Delete
    Completable deleteAuthUser(AuthUser authUser);

    @Query("DELETE FROM authUser_database")
    Completable deleteAllAuthUser();




}

File: KotlinDemoGithubProject/app/src/main/java/com/example/kotlindemogithubproject/RoomDao/AuthUserDao.kt
package com.example.kotlindemogithubproject.RoomDao

import androidx.room.*


@Dao
interface AuthUserDao{
    @Query("SELECT * FROM AUTH_USER WHERE language_id = :id")
    suspend fun getLanguageById(id: Int): AuthUser

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertLanguage(authUser: AuthUser)


    @Update(onConflict = OnConflictStrategy.REPLACE)
    suspend fun updateLanguage(authUser: AuthUser)


    @Query("DELETE FROM AUTH_USER WHERE language_id = :id")
    suspend fun deleteLanguageByID(id:Int)

}
